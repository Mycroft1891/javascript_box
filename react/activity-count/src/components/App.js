import { Component }  from 'react'
import SkiDayCount    from './SkiDayCount'
import SkiDayList     from './SkiDayList'
import AddDayForm     from './AddDayForm'
import Menu           from './Menu'
import MembersList    from './MembersList'
import fetch          from 'isomorphic-fetch'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      allSkiDays: [
        {
          mountain: "Fuji",
          date: "2015-2-15",
          snow: true,
          hike: true
        },
        {
          mountain: "Hartz",
          date: "2012-4-5",
          snow: true,
          hike: false
        },
        {
          mountain: "Reichenbach",
          date: "1891/5/4",
          snow: false,
          hike: true
        }
      ],
      members: [],
      loading: false
    } // state
    this.addDay = this.addDay.bind(this)
  } // constructor

  componentDidMount() {
      this.setState({loading: true})
      fetch('https://api.randomuser.me/?nat=US&results=12')
        .then(response => response.json())
        .then(json => json.results)
        .then(members => this.setState({
          members,
          loading: false
        }))
  }

  addDay(newDay) {
    this.setState({
      allSkiDays: [
        ...this.state.allSkiDays,
        newDay
      ]
    })
  }

  countDays(filter) {
    return this.state.allSkiDays.filter((day) => {
      if (filter) return day[filter]
      return day
    }).length
  }

  render() {
    return(<div className="app">
            <Menu />
            { (this.props.location.pathname === "/")
              ? <SkiDayCount  total={ this.countDays() }
                              snow={ this.countDays("snow") }
                              hike={ this.countDays("hike") } />
              : (this.props.location.pathname === "/add-day")
              ? <AddDayForm onNewDay={ this.addDay }/>
              : (this.props.location.pathname === "/members")
              ? <MembersList members={ this.state.members } />
              :<SkiDayList days={ this.state.allSkiDays } filter={this.props.params.filter}/>}
          </div>)
  }
}

export default App
