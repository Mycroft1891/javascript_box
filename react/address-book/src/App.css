body {
    margin: 0;
    padding: 0;
    background: linear-gradient(to left,
            indigo 0%,
            darkred 100%);
}