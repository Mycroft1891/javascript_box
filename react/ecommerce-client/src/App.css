body {
    font-family: 'Open Sans Condensed';
    padding: 15px;
}

a {
    text-decoration: none;
    color: black;
}

* {
    box-sizing: border-box;
}