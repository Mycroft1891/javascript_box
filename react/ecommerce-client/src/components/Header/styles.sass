.header
    height: 50px
    width: 100%
    display: flex
    justify-content: space-between
    margin-bottom: 20px

    .logo-container
        height: 100%
        width: 50px
        padding: 0 10px

    .menu
        width: 50%
        height: 100%
        display: flex
        align-items: center
        justify-content: flex-end

        .item
            font-size: 20px
            padding: 10px 15px
