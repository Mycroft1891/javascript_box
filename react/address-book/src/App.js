import React, { useEffect, useState } from 'react';
import SearchBar from "./components/Search/SearchBar";
import CardList from "./components/Card/CardList";
import './App.css';

const App = () => {
  const [users, setUsers] = useState([]);
  const [searchValue, setSearchValue] = useState('');

  useEffect(() => {
    (async () => {
      const data = await fetch('https://jsonplaceholder.typicode.com/users');
      const users = await data.json();
      setUsers(users);
    })();
  }, []);

  const handleSearchInput = (event) => {
    setSearchValue(event.target.value.toLowerCase());
  }

  const filteredUsersOnSearch = users.filter(user => (
    user.name.toLowerCase().includes(searchValue)
  ))

  return (
    <div className="App">
      <h1>Addressbuch</h1>
      <SearchBar
        handleSearchInput={handleSearchInput}
        searchValue={searchValue}
      />
      <CardList users={filteredUsersOnSearch} />
    </div>
  );
}

export default App;
