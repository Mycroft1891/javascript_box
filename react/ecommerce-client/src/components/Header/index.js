import React from 'react';
import { Link } from 'react-router-dom';
import { ReactComponent as Logo } from '../../assets/beer.svg';
import './styles.sass';
import { auth } from '../../firebase';

const Header = ({ currentUser }) => (
  <div className="header">
    <Link to="/" className="logo-container">
      <Logo className="logo" />
    </Link>
    <div className="menu">
      <Link to="/shop" className="item">
        Shop
      </Link>
      <Link to="/conctact" className="item">
        Contact
      </Link>
      {currentUser && (
        <Link to="#" className="item" onClick={() => auth.signOut()}>
          Sign Out
        </Link>
      )}
      {!currentUser && (
        <Link to="/signin" className="item">
          Sign In
        </Link>
      )}
    </div>
  </div>
);

export default Header;
