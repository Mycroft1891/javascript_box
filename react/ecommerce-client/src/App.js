import React, { useState, useEffect } from 'react';
import { Switch, Route } from 'react-router-dom';
import Header from './components/header/header';
import Homepage from './pages/homepage/Homepage';
import Shop from './pages/shop/shop';
import Authentication from './pages/authentication/authentication';
import { auth } from './firebase';
import './App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { currentUser: null };
  }

  componentDidMount() {
    auth.onAuthStateChanged((user) => {
      this.setState({ currentUser: user });
    });
  }

  componentWillUnmount() {
    auth.unsubscribeFromAuth();
  }

  render() {
    const { currentUser } = this.state;
    return (
      <div>
        <Header currentUser={currentUser} />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/shop" component={Shop} />
          <Route path="/signin" component={Authentication} />
        </Switch>
      </div>
    );
  }
}

export default App;
