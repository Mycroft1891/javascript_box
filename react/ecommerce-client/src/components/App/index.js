import React, { useState, useEffect } from 'react';
import { Switch, Route } from 'react-router-dom';
import Header from '../Header';
import Homepage from '../../pages/Homepage';
import Shop from '../../pages/Shop';
import Authentication from '../../pages/Authentication';
import { auth, createUserProfileDocument } from '../../firebase';
import './styles.css';

const App = () => {
  const [currentUser, setCurrentUser] = useState({});

  useEffect(() => {
    const unsubscribeFromAuth = auth.onAuthStateChanged(async (userAuth) => {
      if (userAuth) {
        const userRef = await createUserProfileDocument(userAuth);
        userRef.onSnapshot(({ id, data }) => setCurrentUser({ id, ...data() }));
      } else {
        setCurrentUser({ ...userAuth });
      }
    });

    return () => unsubscribeFromAuth();
  }, []);

  return (
    <div>
      <Header currentUser={currentUser} />
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route path="/shop" component={Shop} />
        <Route path="/signin" component={Authentication} />
      </Switch>
    </div>
  );
};

export default App;
