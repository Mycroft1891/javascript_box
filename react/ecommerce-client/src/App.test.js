import React from 'react';
import { shallow } from 'enzyme';
import { Switch, Route } from 'react-router-dom';
import App from './App';
import Header from './components/header/header';

describe('App component', () => {
  let app;

  beforeEach(() => {
    app = shallow(<App />);
  });

  test('renders app component', () => {
    expect(app.debug()).toMatchSnapshot();
  });

  test('currentUser is set to null', () => {
    expect(app.state()).toEqual({ currentUser: null });
  });

  test('renders all sub-components', () => {
    expect(app.find(Header).length).toEqual(1);
    expect(app.find(Switch).length).toEqual(1);
    expect(app.find(Route).length).toEqual(3);
  });
});
