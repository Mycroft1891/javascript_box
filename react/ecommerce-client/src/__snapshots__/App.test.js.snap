// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders app component 1`] = `
"<div>
  <Header currentUser={{...}} />
  <Switch>
    <Route exact={true} path=\\"/\\" component={[Function: Homepage]} />
    <Route path=\\"/shop\\" component={[Function: Shop]} />
    <Route path=\\"/signin\\" component={[Function: Authentication]} />
  </Switch>
</div>"
`;
